src/action.c

```diff
@@ -348,23 +348,29 @@ action_is_valid(struct action *action)
 	return false;
 }
 
-void action_list_free(struct wl_list *action_list)
+void
+action_free(struct action *action)
 {
+	/* Free args */
 	struct action_arg *arg, *arg_tmp;
+	wl_list_for_each_safe(arg, arg_tmp, &action->args, link) {
+		wl_list_remove(&arg->link);
+		zfree(arg->key);
+		if (arg->type == LAB_ACTION_ARG_STR) {
+			free((void *)action_str_from_arg(arg));
```

Comment:
Is this right? Looks odd.
Shouldn't it be `free(arg->value);`

Reply:
This PR just refactored that bit. I agree though that we could cast to `struct action_arg_str` and then free that value instead of "unconst" the return value from `action_str_from_arg()`.

Reply:
Added two variants and opened another review comment to select one of the variants. Resolving this one.